src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java

```diff
@@ -0,0 +1,104 @@
+package com.clustercrew.messagequeue;
+
+import org.apache.bookkeeper.client.*;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import com.clustercrew.messagequeue.MessageQueueOuterClass.Message;
+
+public class BookKeeperClient {
+    private final BookKeeper bookKeeper;
+
+    public BookKeeperClient(String bkServers) throws Exception {
+        ClientConfiguration config = new ClientConfiguration();
+        config.setMetadataServiceUri("zk+null://" + bkServers + "/ledgers");
+        this.bookKeeper = BookKeeper.forConfig(config).build();
+        System.out.println("Connected to BookKeeper: " + bkServers);
+    }
+
+    /**
+     * Creates a new ledger for a topic partition.
+     *
+     * @param topic     The topic name.
+     * @param partition The partition number.
+     * @throws Exception If an error occurs while creating the ledger.
+     */
+    public void createLedger(String topic, int partition) throws Exception {
+        try (LedgerHandle ledger = bookKeeper.createLedger(
```

Comment:
Why are we not using topic and partition to createLedger here?

Reply:
Resolved in commit adf681753297ff834cea8db332359d743fd8bc1a "Resolved a minor bug in BookKeeperClient"